### DISCUSSION TITLE
mapStores and nested components

### DISCUSSION THREAD

**Discussion by plweil** [vuejs/pinia#1136] [Help and Questions]
I'm using `mapStores()` in my root-level component, but I'm finding that I need to map one or more of these stores again in some of my nested components. I thought `mapStores()` would work for all nested components. What determines when `mapStores()` will no longer apply to a nested component? So far I've been using trial-and-error, but it would be helpful to know when I need to map relevant stores a second time, or what I can do to have an initial `mapStores()` work in all of my nested components (we're talking about maybe 5 nested components, not dozens).

Related question: can someone please verify whether using `mapStores()` should mitigate the need to use `mapWritableState()`, or do I need to apply the latter explicitly whenever I need to write to state properties? My initial experience suggests that I need to use `mapWritableState()` regardless.

**Comment by 3zzy** [comment]
I have the same question. I was hoping to use the `mapStores()` in the root component and somehow `provide` it from root and `inject` the store into the child components but that doesn't seem to work. Any ideas @posva ? 
Thanks!
